dashboard/engines/marketing/app/controllers/marketing/teacher/promotions_controller.rb

```diff
@@ -16,6 +16,7 @@ def show
         result = entry.fields.clone
         ads = result[:sidebar_ads].map do |ad|
           ad.fields.transform_values! {|v| v.is_a?(Contentful::Asset) ? v.image_url : v}
+          ad.fields.merge!(id: ad.id)
```

Comment:
In retrospect I don't know why I used the `!` version of `transform_values` here. I think we can correct that and also chain the merge like this:
```suggestion
          ad.fields.transform_values {|v| v.is_a?(Contentful::Asset) ? v.image_url : v}.merge(id: ad.id)
```